### ISSUE TITLE
[BUG] Operation not allowed: `NOT NULL in ALTER TABLE ADD COLUMNS` is not supported for Delta tables

### ISSUE THREAD

**Issue by li-boxuan** [delta-io/delta#3360]
I understand this might be by design, but https://docs.delta.io/latest/delta-batch.html#add-columns could be made more clear. It currently says,

> By default, nullability is true.

which gives me an impression that one can override the nullability, but I got

```
spark-sql (default)> ALTER TABLE basic ADD COLUMNS (col1_nonnull2 integer NOT NULL);
[DELTA_OPERATION_NOT_ALLOWED] Operation not allowed: `NOT NULL in ALTER TABLE ADD COLUMNS` is not supported for Delta tables
```

Version:

```
        :: modules in use:
        io.delta#delta-spark_2.12;3.1.0 from central in [default]
        io.delta#delta-storage;3.1.0 from central in [default]
        org.antlr#antlr4-runtime;4.9.3 from central in [default]
```

Related issue: #831

**Comment by rishabhchaudha** [comment]
I've noticed that an error is intentionally thrown when specifying 'Not Null' in this code. This decision should be by design , especially during schema alterations where the default value for new columns is logically null.

<img width="671" alt="image" src="https://github.com/user-attachments/assets/f3d5c185-532b-4673-b111-b2f742898275">

**Comment by li-boxuan** [comment]
> This decision should be by design , especially during schema alterations where the default value for new columns is logically null.

Yeah I agree with that. I would consider it as a glitch in the doc to be improved.

**Comment by MrPowers** [comment]
@li-boxuan - thanks for creating this issue.  Would you like to submit a pull request to update the language of the docs?

**Comment by foss-contributor** [comment]
@li-boxuan, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
